scripts/nlp_language_modeling/convert_hf_falcon_to_nemo.py

```diff
@@ -0,0 +1,294 @@
+    }
+
+    mcore_customization_config_dict = {
+        "new_decoder_architecture": falcon_config.new_decoder_architecture,
+        "parallel_attention": falcon_config.parallel_attn,
+    }
+
+    tokenizer_dict = {
+        "library": "huggingface",
+        "type": args.input,
+        "use_fast": True,
+    }
+    trainer_dict = {
+        "devices": 1,
+        "num_nodes": 1,
+        "accelerator": "gpu" if args.cuda else "cpu",
+        "precision": args.precision,
+        "logger": False,
+        "enable_checkpointing": False,
+        "max_epochs": -1,
+        "max_steps": 100000,
+        "log_every_n_steps": 10,
+        "val_check_interval": 100,
+        "limit_val_batches": 50,
+        "limit_test_batches": 500,
+        "accumulate_grad_batches": 1,
+        "gradient_clip_val": 1.0,
+        "benchmark": False,
+        "enable_model_summary": False,
+        "strategy": NLPDDPStrategy(),
+    }
+
+    # Additional logic for position_embedding_type = alibi
+    if falcon_config.alibi:
+        try:
+            raise ValueError(
+                "Alibi is not yet supported in Megatron Core, \
+                force to use RoPE will generate suboptimal responses"
+            )
+        except ValueError as e:
+            print(e)
+
+    # Additional logic for num_query_groups
+    if override_model_dict.get("num_query_groups") is None:
+        if falcon_config.new_decoder_architecture:
+            override_model_dict["num_query_groups"] = falcon_config.num_kv_heads
+        elif falcon_config.multi_query:
+            override_model_dict["num_query_groups"] = 1
+
+    # Additional logic for bias fusion
+    if falcon_config.bias:
+        override_model_dict["bias_activation_fusion"] = True
+        override_model_dict["bias_dropout_add_fusion"] = True
+
+    # Addtional logic for rope scaling
+    if falcon_config.rope_scaling is not None:
+        if falcon_config.rope_scaling.type == 'linear':
+            override_model_dict['seq_len_interpolation_factor'] = falcon_config.rope_scaling.factor
+        else:
+            raise ValueError("Only linear rope scaling type is supported now")
+
+    model_dict.update(override_model_dict)
+    model_dict["tokenizer"] = tokenizer_dict
+    model_dict["name"] = 'megatron_falcon_gpt'
+    model_dict["mcore_customization_config"] = mcore_customization_config_dict
+
+    omega_cfg = OmegaConf.create(model_dict)
+
+    trainer = pl.Trainer(**trainer_dict)
+
+    logging.info("Creating Megatron model...")
+    tik = time.time()
+    model = MegatronGPTModel(omega_cfg, trainer)
+
+    logging.info("Loading HuggingFace model...")
+    model_hf = AutoModelForCausalLM.from_pretrained(args.input)
+
+    state_dict_hf = model_hf.state_dict()
+    convert_dict = convert_state_dict(state_dict_hf, amp=omega_cfg.megatron_amp_O2)
+
+    logging.info("Loading state dict...")
+    missing_keys, unexpected_keys = model.load_state_dict(convert_dict, strict=False)
```

Comment:
Would you be fine with defining a "load_state_dict_helper" analogously as Im doing here in another MR https://github.com/NVIDIA/NeMo/pull/7935/commits/ad4d8909e1de041adbab42a08c2aa70ee70a9d6e (see bottom part) so that we could later reuse it for different HF models (Llama, StarCoder, etc.)?

Just a defining and using it will do - applying it to other models should be done independently in subsequent MRs.